rita_client/src/operator_update/ops_websocket.rs

```diff
@@ -0,0 +1,118 @@
+use std::{str, thread, time::Duration};
+
+use actix_async::System;
+use actix_web_actors::ws;
+use althea_types::OperatorUpdateMessage;
+use awc::ws::Frame;
+use futures::{SinkExt, StreamExt};
+use tokio::time::timeout;
+
+use crate::operator_update::{get_operator_checkin_data, handle_operator_update};
+
+/// This function spawns a thread solely responsible for performing the websocket operator update
+pub fn start_operator_socket_update_loop() {
+    // first actually get the websocket address
+    info!("Starting operator socket update loop");
+    send_websocket_update();
+}
+
+// How long we wait between checkins with the operator server
+const SOCKET_UPDATE_FREQUENCY: Duration = Duration::from_secs(10);
+
+/// Send an update to ops server via websocket
+pub fn send_websocket_update() {
+    let url: &str;
+    if cfg!(feature = "dev_env") {
+        url = "http://7.7.7.7:8080/ws/";
+    } else if cfg!(feature = "operator_debug") {
+        url = "http://192.168.1.214:8080/ws/";
+    } else {
+        url = "https://operator.althea.net:8080/ws/";
+    }
+    thread::spawn(move || loop {
+        let runner = System::new();
+        runner.block_on(async move {
+            let res = awc::Client::new().ws(url).connect().await;
+            match res {
+                Ok((res, mut ws)) => {
+                    info!("Websocket actor is connected {:?}", res);
+                    let mut ops_last_seen_usage_hour: Option<u64> = None;
+                    loop {
+                            // check if there is anything to read first
+                            let res = timeout(Duration::from_millis(10), ws.next()).await;
```

Comment:
this loop should be split out into its own function so it's easier to separate the logic form all the threads/runners